subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/BuildEnvironmentReportTaskIntegrationTest.groovy

```diff
@@ -15,12 +15,30 @@
  */
 package org.gradle.api.tasks.diagnostics
 
+import com.google.common.base.StandardSystemProperty
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.internal.jvm.Jvm
+import org.gradle.internal.jvm.inspection.JvmVendor
 import org.gradle.test.fixtures.file.LeaksFileHandles
 
+import java.util.regex.Pattern
+
 class BuildEnvironmentReportTaskIntegrationTest extends AbstractIntegrationSpec {
-    def setup() {
-        executer.requireOwnGradleUserHomeDir()
+    def "reports Build JVM information"() {
+        when:
+        run(":buildEnvironment")
+
+        then:
+        // Not asserting over the exact output, just that important info is printed
+        JvmVendor currentVendor = JvmVendor.fromString(StandardSystemProperty.JAVA_VM_VENDOR.value())
+        output.matches(
+            "(\n|.)*Build JVM: ${Pattern.quote(currentVendor.displayName)} .*${Pattern.quote(StandardSystemProperty.JAVA_VM_VERSION.value())}.*\\n" +
+                ".*Location:\\s+${Pattern.quote(StandardSystemProperty.JAVA_HOME.value())}.*\\n" +
+                ".*Language Version:\\s+${Jvm.current().javaVersionMajor}.*\\n" +
+                ".*Vendor:\\s+${Pattern.quote(currentVendor.displayName)}.*\\n" +
+                ".*Architecture:\\s+${Pattern.quote(StandardSystemProperty.OS_ARCH.value())}.*\\n" +
+                ".*Is JDK:\\s+${Jvm.current().jdk}.*(\\n|.)*"
```

Comment:
Instead of trying to test this exhaustively here, could we instead....

1. Check that there's a build JVM info block in buildEnvironment
2. Check that the Location is seen

I think ToolchainReportRendererTest already checks the format of the message.